### PR TITLE
fix: Community logo and title animation

### LINKED ISSUE #19635: Community logo and title stucks within banner
Was fixed recently in https://github.com/status-im/status-mobile/pull/19395 but turned out that is still reproducible cc @ajayesivan 

Don't have clear reprduction steps. Sometimes community logo and title is displayed within banner after opening community screen. See video below. 

photo_2024-04-15 13 52 57

https://github.com/status-im/status-mobile/assets/97245802/98f588be-d0b4-4c77-8ae6-714b01ac0dc4

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by ajayesivan**
fixes #19635 

#### Areas that maybe impacted
- Community overview screen

### Steps to test
- Open any community and check the header (The issue was random, so better to check with different communities & multiple times)

### Before and after screenshots comparison

Before | After
-|-
 | 

status: ready

### CODE REVIEW

File: src/quo/components/navigation/page_nav/view.cljs [hunk 1]
```diff
@@ -251,86 +252,95 @@
...
-       :account-switcher account-switcher}]]
-
-    :title
-    (let [centered? (= text-align :center)]
+  (let [center-content-container-style (reanimated/apply-animations-to-style
+                                        (if center-opacity
+                                          {:opacity center-opacity}
+                                          nil)
+                                        (style/center-content-container
+                                         (if (= type :title) (= text-align :center) false)))
```
Comment (Parveshdhull) [inline src/quo/components/navigation/page_nav/view.cljs]: `center-content-container (and (= type :title) (= text-align :center))`?

**Review by Parveshdhull — APPROVED** [review]

File: src/quo/components/navigation/page_nav/view.cljs (hunk 1 above)
Comment (seanstrom) [inline src/quo/components/navigation/page_nav/view.cljs]: Hey @ajayesivan 👋 Thanks for the PR 🙏 

I have some questions around this refactor, and I want to make sure I understand the code:
* I was noticing that some components like `wallet-networks-center` and `dropdown-center` were always passing `centered?` as `true`. Should they continue to do that?
* I was also noticing that most of the other components like `channel-center` always passed `centered?` as `false`. And that `title-center` was the only one that checked for `(= text-align :center)`. Should they all now only be centered based on text-align?

I suppose it makes sense that we have a standard way of determining when something should be centered, though I wonder if the code had some special assumptions by always passing `false` or `true`.

**Comment by ulisesmac** [comment]
> Steps to test

>    Open any community and check the header (The issue was random, so better to check with different communities & multiple times)


Hi @ajayesivan 

Did you find the root reason for this bug to happen?
If so, could you please explain it?

From what can I see and since I'm not familiar with this component:
I guess the community header should only appear when you scroll, if so, then maybe the problem is `flat-list` reporting the scroll position (via the `on-scroll` event), so please make sure we are using the `reanimated/flat-list` version instead of the `rn/flat-list`.

Another cause might be that it takes some time to render or fetch the content of the community, and the flat list is shorter at the beginning so we display the header.

File: src/quo/components/navigation/page_nav/view.cljs (hunk 1 above)
Comment (ajayesivan) [reply src/quo/components/navigation/page_nav/view.cljs]: Hi @seanstrom, yes, it is a bit confusing and maybe not the ideal solution for this. But first of all, I'm not trying to fix that in this PR. My goal in this PR is to fix the animation.

In this component, only the type `title` has two alignments: `center` and `left`. So the `text-align` property is only considered when the type is `title`.

Design: https://www.figma.com/design/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=159-6028&m=dev

**Comment by ajayesivan** [comment]
Hi @ulisesmac, in this component, we were using the `center-opacity` prop passed down from the parent component (`scroll-page`) for animation. We were using `center-opacity` directly in the style despite it being a shared value. In this PR, I have updated the code to use the 'use-shared-styles' hook instead.

**Review by ulisesmac — APPROVED** [review]

**Comment by mariia-skrypnyk** [comment]
Hi @ajayesivan !

Thanks for your fix.
Looks good on both plaforms.

E2E are known and not related.
Please, merge your PR!

**Pull request merged**